paasta_tools/cli/cmds/spark_run.py

```diff
@@ -537,7 +579,14 @@ def get_spark_app_name(original_docker_cmd: Union[Any, str, List[str]]) -> str:
     if spark_app_name is None:
         spark_app_name = "paasta_spark_run"
 
-    spark_app_name += f"_{get_username()}"
+    spark_app_name += "_{}".format(get_username())
+    if enable_k8s_autogen:
+        document = POD_TEMPLATE.format(spark_app_name=spark_app_name)
+        parsed_pod_template = yaml.load(document)
+        f = open(POD_TEMPLATE_PATH, "w")
+        f.write(yaml.dump(parsed_pod_template))
+        f.close()
```

Comment:
nit: in Python `with open(filename, ...):` is preferred, so that you don't need to write `f.close()` afterwards (the context takes care of it. In a more Pythonic way:
```python
with open(POD_TEMPLATE_PATH, 'w') as f:
    yaml.dump(parsed_pod_template, f)
```